src/System.Threading.Channels/tests/ChannelTestBase.netcoreapp.cs

```diff
@@ -152,7 +150,6 @@ public async Task ReadAllAsync_MultipleEnumerationsToEnd(bool sameEnumerable)
 
             IAsyncEnumerable<int> enumerable = c.Reader.ReadAllAsync();
             IAsyncEnumerator<int> e = enumerable.GetAsyncEnumerator();
-            Assert.Same(enumerable, e);
```

Comment:
If we were previously asserting sameness, is it worth asserting not-sameness now?

Reply:
I'd like for them to be same again in the future, and it's ok if they are now.  We just lack the language feature that let's us both get the sameness and use an async iterator.